validation_pipeline.ipynb

```diff
@@ -555,71 +576,145 @@
       "metadata": {
-        "id": "WpHxY_5rV2er"
+        "id": "_oHdtxS7tT5z"
       },
       "execution_count": null,
       "outputs": []
     },
     {
       "cell_type": "code",
       "source": [
-        "auc_scores = {}\n",
-        "\n",
-        "for radius, fake_sample in fake_samples.items():\n",
-        "  test_dataset = pd.concat([real, fake_sample], axis=1, join='outer')\n",
-        "  test_dataset = real.append(fake_sample, ignore_index=True)\n",
-        "\n",
-        "  predictions = hypothesis.get_predictions(test_dataset,\n",
-        "    isotope_column_names,\n",
-        "    mean_isoscapes,\n",
-        "    vars_isoscapes,\n",
-        "    5)\n",
-        "\n",
-        "  predictions.dropna(subset=['fraud', 'fraud_p_value'], inplace=True)\n",
-        "\n",
-        "  y_true = predictions['fraud']\n",
-        "  # Fraud p value is lower the more positive a prediction/label is.\n",
-        "  # Inverting it gives us the probability of positive label class (fraud).\n",
-        "  y_pred = 1 - predictions['fraud_p_value']\n",
-        "\n",
-        "  precision, recall, thresholds = precision_recall_curve(y_true, y_pred)\n",
-        "\n",
-        "  plt.plot(recall, precision, label=\"model\")\n",
-        "  plt.plot(np.linspace(0, 1, 100), np.ones(100) * 0.5, \"--\", label=\"random\")\n",
+        "def plot_isoscape_precision_recall(\n",
+        "    precision: list[float],\n",
+        "    recall: list[float],\n",
+        "    label: str):\n",
+        "  plt.plot(recall, precision, label=label)\n",
         "  plt.xlabel(\"Recall\")\n",
         "  plt.ylabel(\"Precision\")\n",
         "  plt.title(f\"Precision-Recall curve with Max Fraud Radius of {radius} km\")\n",
-        "  plt.legend()\n",
-        "  plt.show()\n",
-        "\n",
-        "  #Adjusted the shape of precision and recall, since they have ndarray of shape (n_thresholds + 1,)\n",
+        "  plt.legend()"
+      ],
+      "metadata": {
+        "id": "xKlw5wsyuN4n"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def find_p_value(\n",
+        "    precision: list[float],\n",
+        "    recall: list[float],\n",
+        "    thresholds: list[float],\n",
+        "    precision_target: float,\n",
+        "    recall_target: float) -> list[float]:\n",
         "  assert(precision_target or recall_target)\n",
         "  if precision_target:\n",
         "    target_pos = np.argwhere(precision[:-1] >= precision_target)\n",
         "  else:\n",
         "    target_pos = np.argwhere(recall[:-1] >= recall_target)\n",
+        "  # No precision/recall is greater than or equal to the target\n",
+        "  if len(target_pos) < 1:\n",
+        "    if precision_target:\n",
+        "      target_pos = [np.argmax(precision[:-1])]\n",
+        "    else:\n",
+        "      target_pos = [np.argmax(recall[:-1])]\n",
+        "  print(target_pos)\n",
```

Comment:
Is this just a debug statement or a useful output?

Reply:
Debug. Removed.